src/compas_model/elements/column.py

```diff
@@ -0,0 +1,231 @@
+        self.polygon_bottom, self.polygon_top = self.compute_top_and_bottom_polygons()
+
+    # =============================================================================
+    # Implementations of abstract methods
+    # =============================================================================
+
+    def compute_aabb(self, inflate: float = 0.0) -> Box:
+        """Compute the axis-aligned bounding box of the element.
+
+        Parameters
+        ----------
+        inflate : float, optional
+            The inflation factor of the bounding box.
+
+        Returns
+        -------
+        :class:`compas.geometry.Box`
+            The axis-aligned bounding box.
+        """
+        points: list[list[float]] = self.geometry.vertices_attributes("xyz")  # type: ignore
+        box: Box = Box.from_bounding_box(bounding_box(points))
+        box.xsize += inflate
+        box.ysize += inflate
+        box.zsize += inflate
+        return box
+
+    def compute_obb(self, inflate: float = 0.0) -> Box:
+        """Compute the oriented bounding box of the element.
+
+        Parameters
+        ----------
+        inflate : float, optional
+            The inflation factor of the bounding box.
+
+        Returns
+        -------
+        :class:`compas.geometry.Box`
+            The oriented bounding box.
+        """
+        points: list[list[float]] = self.geometry.vertices_attributes("xyz")  # type: ignore
+        box: Box = Box.from_bounding_box(oriented_bounding_box(points))
+        box.xsize += inflate
+        box.ysize += inflate
+        box.zsize += inflate
+        return box
+
+    def compute_collision_mesh(self) -> Mesh:
+        """Compute the collision mesh of the element.
+
+        Returns
+        -------
+        :class:`compas.datastructures.Mesh`
+            The collision mesh.
+        """
+        from compas.geometry import convex_hull_numpy
+
+        points: list[list[float]] = self.geometry.vertices_attributes("xyz")  # type: ignore
+        vertices, faces = convex_hull_numpy(points)
+        vertices = [points[index] for index in vertices]  # type: ignore
+        return Mesh.from_vertices_and_faces(vertices, faces)
+
+    def compute_elementgeometry(self) -> Mesh:
+        """Compute the shape of the column from the given polygons.
+        This shape is relative to the frame of the element.
+
+        Returns
+        -------
+        :class:`compas.datastructures.Mesh`
+
+        """
+
+        offset: int = len(self.polygon_bottom)
+        vertices: list[Point] = self.polygon_bottom.points + self.polygon_top.points  # type: ignore
+        bottom: list[int] = list(range(offset))
+        top: list[int] = [i + offset for i in bottom]
+        faces: list[list[int]] = [bottom[::-1], top]
+        for (a, b), (c, d) in zip(pairwise(bottom + bottom[:1]), pairwise(top + top[:1])):
+            faces.append([a, b, d, c])
+        mesh: Mesh = Mesh.from_vertices_and_faces(vertices, faces)
+        return mesh
```

Comment:
why is a column not simply a box and its computed geometry not simply box to mesh?